Style updates for the layers panel

designedbyscience opened the pull request:
Followed @placegraphichere design. He will be doing additional icon work.

Panel Columns are a little wider than before

Groups and Artboard layers are supposed to have a lighter/easier to find icon.

This does not include warn/error states for smart objects (need to talk to @volfied about those)

I made an adjustment to how the layers/libraries panels interact (a single one open looks better, and both open split the space evenly).

There are also changes to address #1368 and #1853

@placegraphichere Please review this

iwehrman commented:
I'm not sure what my constructive suggestion here is, but the vertical spacing feels weird at the intersection of the header and the layers panel: 
![vert](https://s3.amazonaws.com/f.cl.ly/items/0M0W1O152E0y16292O1k/Image%202015-08-04%20at%203.07.44%20PM.png)

iwehrman commented:
I think the names of layers hidden on canvas are way too hard to read due to lack of contrast:
![contrast](https://s3.amazonaws.com/f.cl.ly/items/0b1X2x2X0F1H1P172S33/Image%202015-08-04%20at%203.10.22%20PM.png)

iwehrman commented:
Again, I'm not sure what to suggest, but the transition from one vertical panel with a scroll gutter to one without looks weird:
![gutter](https://s3.amazonaws.com/f.cl.ly/items/26023Z0y2K3w3q3X2y1Y/Image%202015-08-04%20at%203.14.12%20PM.png)

iwehrman commented:
All the datalist menus are misaligned:
![menu](https://s3.amazonaws.com/f.cl.ly/items/2v3h0d1G0c2u430U173m/Image%202015-08-04%20at%203.15.52%20PM.png)

iwehrman commented:
Maybe this is on the roadmap, but I think that we should get rid of the little panel-hover arrows to collapse the columns since we've got the buttons. I might still be in favor of a similar panel-hover arrow thing to _swap_ the columns though.

iwehrman commented:
Color picker is misaligned too:
![color](https://s3.amazonaws.com/f.cl.ly/items/1S1S1k2h00372L35003H/Image%202015-08-04%20at%203.19.03%20PM.png)

iwehrman commented:
In the special case of two-panel columns like Layers/Libraries, it seems like collapsing one row panel should always expand the other one: 
![expand](https://s3.amazonaws.com/f.cl.ly/items/462C3m211Y2I0K310g1w/Image%202015-08-04%20at%203.19.51%20PM.png)

iwehrman commented:
The layers/libraries column bounces a bit when opening and closing the styles/export column: 
![bounce](https://s3.amazonaws.com/f.cl.ly/items/1V1g0p0L1a3g3u191g23/Screen%20Recording%202015-08-04%20at%2004.41%20PM.gif)

iwehrman commented:
This isn't your fault, but I think it's time for the ALL-CAPS tooltips to go:
![caps](https://s3.amazonaws.com/f.cl.ly/items/290j313P3p3b221X0c0d/Image%202015-08-04%20at%204.44.26%20PM.png)

placegraphichere commented:
For the look of the scrollbar and panel vertical alignment:  Unless we go to a JS library for scrolling, I think we should minimize the width to 0.4rem, and remove the background color of the scrollbar. This looks a bit better. @designedbyscience had concerns the width will not work on WIN, which may be a problem for another day when I can get my VM running.

![screen shot 2015-08-06 at 3 17 08 pm](https://cloud.githubusercontent.com/assets/3229167/9122393/574c0932-3c4e-11e5-8cbe-91f16b87b798.png)

placegraphichere commented:
Cases where the user tries to close the last possible panel, and could arrive in a state where no panels are open in a column. We'd like to avoid this. One option is to simply close the entire column, however that seems a little unexpected to me. 

Instead, I think we should just not allow the user to close the last open panel. This will help standardize the rules for both the Properties column (3+ panels) and the Layers column (2 panels). 

The down side to just "opening something else instead" would be the Properties column, we'd need to have a preferred "open" panel in a case where the user closes the last panel—and what would we open instead? Instead, in this proposal we're keeping the same panel functionality for both columns, under one simple rule.

Logic might look like:
`Panel Double-Click = Am I the last panel open ? Do nothing : Close this panel;`

placegraphichere commented:
Icon and hidden layer contrast: we should up the value of the text to @label-secondary-color: #545454;. For the icons, I believe that has been improved. We should also get a better contrast with the icons once the new icons are in place (aka defer).

designedbyscience commented:
Lots o' updates, addressed above concerns and bugs.

I have left the little black "Hide this column" arrows for the moment

The behavior for the panels collapsing is as Seth described above. We can test the different versions when we have more panels in there

@iwehrman Back to you

iwehrman commented:
Needs a rebase.

designedbyscience commented:
@iwehrman rebased

iwehrman commented:
Is it intentional that the hairlines at the top and the bottom of the libraries panel are different colors?

iwehrman commented:
Without the underline, the floating down arrow is a little mysterious:
![image](https://cloud.githubusercontent.com/assets/1075154/9209832/7112d336-402d-11e5-9c6a-ea2037c70b15.png)

iwehrman commented:
Eh, I just realized I'm complaining about the wrong panel. I'll move those complaints elsewhere. SORRY.

iwehrman commented:
Actually about layers panel: the layer drag style is too short horizontally. As soon as you start the drag, the layers are reduced in width by ~25%.

iwehrman commented:
Bad text color in layer rename mode:
![image](https://cloud.githubusercontent.com/assets/1075154/9209903/f1d64bd8-402d-11e5-870f-9db426254efb.png)

iwehrman commented:
Also, is this still the desired text selection color? 
![image](https://cloud.githubusercontent.com/assets/1075154/9209919/05ea966a-402e-11e5-8dbb-f1266d04d6bb.png)

iwehrman commented:
Error when attempting to collapse the libraries panel:
```
Uncaught TypeError: Cannot read property 'parentElement' of null PanelSet.jsx:140
 React.createClass._handlePanelVisibilityToggle react-with-addons.js:3311
 executeDispatch react-with-addons.js:17428
 SimpleEventPlugin.executeDispatch react-with-addons.js:3299
...
```

iwehrman commented:
Look who I found!
![image](https://cloud.githubusercontent.com/assets/1075154/9210454/439b7940-4031-11e5-91be-193ecc7d192d.png)

designedbyscience commented:
You win the prize!

iwehrman commented:
Also, something is screwed up with `ui.togglePinnedToolbar`. It never finishes executing:
```
Enqueuing action ui.togglePinnedToolbar; 0/0
Executing action ui.togglePinnedToolbar after waiting 0ms; 1/0
Enqueuing action preferences.setPreference; 1/0
Enqueuing action documents.createNew; 1/1
Enqueuing action documents.createNew; 1/2
Enqueuing action documents.createNew; 1/3
Enqueuing action documents.createNew; 1/4
```

iwehrman commented:
He's over here too:
![image](https://cloud.githubusercontent.com/assets/1075154/9210520/9bbc2de0-4031-11e5-9584-9adfad7a47b0.png)

iwehrman commented:
Gotta say I still think the contrast is still too low here:
![image](https://cloud.githubusercontent.com/assets/1075154/9210542/c5cf1502-4031-11e5-8d8c-1f5621d1bc59.png)

iwehrman commented:
`ui.updatePanelSizes` is really slow, and seems to get called twice in a row quite often. Anything we can do about that?

iwehrman commented:
Expanding or collapsing layers does not cause the tab bar to resize the document titles. Instead, it seems like they only resize when the window is resized.

iwehrman commented on src/js/jsx/sections/layers/LayersPanel.jsx:
```diff
@@ -595,14 +595,6 @@ define(function (require, exports, module) {
-                <div
-                    title={strings.TOOLTIPS.LAYER_COUNT}
-                    className="layer-count">
-                    {doc.layers.selected.size}<span className="text-fancy"> oƒ </span>{doc.layers.count}
```
I like this thing, but oh well. :cocktail:

iwehrman replied:
We could probably rip out the associated `count` derived property, since I'm pretty sure it was only being used there.

iwehrman commented on src/js/util/svg.js:
```diff
@@ -46,6 +46,10 @@ define(function (require, exports) {
             iconID += layer.kind;
         }
 
+        if ((layer.isArtboard || layer.layerKinds.GROUP) && !layer.expanded) {
```
Artboard layers are groups, so I think the first disjunct is redundant.

iwehrman commented:
OK, done with a second pass. Beyond the usual nits, the main concern is the `ui` action weirdness and performance issues. Happy to chat more offline. Other than that, it's obviously getting close. Everyone should be prepared to hold onto their butts.

designedbyscience commented:
1. Drag Style is not wide enough - Now it is wide enough
2. Bad text color - new text color
3. Selection color - keeping same color for now
4. Errors collapsing Library Panel - Fixed that
5. Borders on document header - now only shown in withdoc state
6. Pinned Toolbar errors - Fixed now
7. Contrast - adjusted a litte more. However, we may need to reconsider the normal text color in order to get more contrast
8. updatePanelSizes - now called just once
9. Document Title - this is a known bug (1890)
10. Count property - removed
11. Fixed expanded icon bug

Okay, back to @iwehrman

iwehrman commented:
Looks great. I filed a few follow-on bugs which I'm sure you'll be excited to address in your next pass. Bombs away!
